components/style/properties/shorthand/position.mako.rs

```diff
@@ -138,6 +138,107 @@
+        let end = if input.try(|i| i.expect_delim('/')).is_ok() {
+            GridLine::parse(context, input)?
+        } else {
+            let mut line = GridLine::default();
+            line.ident = start.ident.clone();       // ident from start value should be taken
+            line
+        };
+
+        Ok(Longhands {
+            grid_${kind}_start: start,
+            grid_${kind}_end: end,
+        })
+    }
+
+    impl<'a> ToCss for LonghandsToSerialize<'a> {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            self.grid_${kind}_start.to_css(dest)?;
+            let ref other = self.grid_${kind}_end;
+            if other.ident.is_some() || other.integer.is_some() {       // don't serialize auto
+                dest.write_str(" / ")?;
+                other.to_css(dest)?;
+            }
+
+            Ok(())
+        }
+    }
+</%helpers:shorthand>
+% endfor
+
+<%helpers:shorthand name="grid-area"
+                    sub_properties="grid-row-start grid-row-end grid-column-start grid-column-end"
+                    spec="https://drafts.csswg.org/css-grid/#propdef-grid-area"
+                    products="gecko">
+    use values::specified::GridLine;
+    use parser::Parse;
+
+    // The code is the same as `grid-{row,column}` except that this can have four values at most.
+    pub fn parse_value(context: &ParserContext, input: &mut Parser) -> Result<Longhands, ()> {
+        let row_start = input.try(|i| GridLine::parse(context, i))?;
+        let (column_start, row_end, column_end) = if input.try(|i| i.expect_delim('/')).is_ok() {
+            let column_start = GridLine::parse(context, input)?;
+            let (row_end, column_end) = if input.try(|i| i.expect_delim('/')).is_ok() {
+                let row_end = GridLine::parse(context, input)?;
+                let column_end = if input.try(|i| i.expect_delim('/')).is_ok() {
+                    GridLine::parse(context, input)?
+                } else {        // grid-column-end has not been given
+                    let mut line = GridLine::default();
+                    line.ident = column_start.ident.clone();        // use ident from grid-column-start
+                    line
+                };
+
+                (row_end, column_end)
+            } else {        // grid-row-start and grid-column-start has been given
+                let mut row_end = GridLine::default();
+                row_end.ident = row_start.ident.clone();
+                let mut column_end = GridLine::default();
+                column_end.ident = column_start.ident.clone();
+                (row_end, column_end)       // use idents from their partners
+            };
+
+            (column_start, row_end, column_end)
+        } else {        // only grid-row-start is given
+            let mut line = GridLine::default();
+            line.ident = row_start.ident.clone();       // use the ident from grid-row-start
```

Comment:
The `<custom-ident>` rule applies to this property as well. These should be set to first ident if only grid-row-start is  `<custom-ident>`.(and applies to above) Otherwise, they should be auto.